src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create new book = " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        Book book = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = createBook(resultSet);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not to find the book by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> booksList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = createBook(resultSet);
+                booksList.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not to find any book in DB", e);
+        }
+        return booksList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setObject(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRow = statement.executeUpdate();
+            if (affectedRow != 0) {
+                return book;
+            } else {
+                throw new DataProcessingException("Expected updated the book"
+                        + book + " But it's not happened");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update the book " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not to delete the book by id = " + id, e);
+        }
+    }
+
+    private Book createBook(ResultSet resultSet) throws SQLException {
```

Comment:
You probably should change this name. You not only create (some abstract) book but also fill it with some specific data.